mitodl/open-discussions
# Added pagination for frontpage and posts APIs and UI

#### What are the relevant tickets?
Fixes #77 
Fixes #199 

#### What's this PR do?
Adds pagination for posts-related things

#### How should this be manually tested?
- Set `OPEN_DISCUSSIONS_CHANNEL_POST_LIMIT` to a low value (e.g. `5`). This controls the max number of posts that appear per page
- Generate enough posts in a channel to trigger pagination
- Go to the front page and ensure you can page in both directions
- Go to the channel you created the posts in and ensure you can page in both directions

#### Any background context you want to provide?
A few known issues:
- If you page back to the first page, you'll still see pagination params in the URL.
- If you have exactly a multiple of the posts per page in a channel, you'll see a "Next" link when you're actually on the last page, which will navigate you to a blank "empty channel" page with no way to get back (aside from back button). 
- If you have any posts by deleted users, you'll see pages with less posts than the limit. You could also see a last page similar to the post above but is actually caused by deleted items.

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
![Pagination](https://media.giphy.com/media/XqpLGIQ7nMtqM/giphy.gif)
